### ISSUE TITLE
Unable to install core_kernel

### ISSUE THREAD

**Issue by ag999** [janestreet/core_kernel#81]
I was trying to install the async library and as a part of many other things, it required the installation of core_kernel. But the installation of core_kernel fails for me. When i run `opam install core_kernel` it fails with this error message `The compilation of core_kernel failed at "jbuilder build -p core_kernel -j 4".`

I am running a Mac (High Sierra 10.13.4). I am running opam version 1.2.2 and compiler 4.06.0. Thank you for the help.

**Comment by SantiagoBautista** [comment]
I got the exact same error on Ubuntu with a freshly installed 4.07.0 ocaml compiler.

Here is the entire error message:
```
=-=- Processing actions -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
[ERROR] The compilation of core_kernel failed at "jbuilder build -p
        core_kernel -j 4".

#=== ERROR while installing core_kernel.v0.11.0 ===============================#
# opam-version 1.2.2
# os           linux
# command      jbuilder build -p core_kernel -j 4
# path         /home/santi/.opam/4.07.0/build/core_kernel.v0.11.0
# compiler     4.07.0
# exit-code    1
# env-file     /home/santi/.opam/4.07.0/build/core_kernel.v0.11.0/core_kernel-2688-d2c37b.env
# stdout-file  /home/santi/.opam/4.07.0/build/core_kernel.v0.11.0/core_kernel-2688-d2c37b.out
# stderr-file  /home/santi/.opam/4.07.0/build/core_kernel.v0.11.0/core_kernel-2688-d2c37b.err
### stderr ###
# Error: The implementation src/info.pp.ml
# [...]
#            val bin_read_t : t Bin_prot.Read.reader
#            val bin_reader_t : t Bin_prot.Type_class.reader
#            val bin_t : t Bin_prot.Type_class.t
#            val sexp_of_t : t -> Ppx_sexp_conv_lib.Sexp.t
#            val to_info : Base.Info.Internal_repr.t -> Base.Info.t
#            val of_info : Base.Info.t -> Base.Info.Internal_repr.t
#          end
#        is not included in
#          (module Base.Info.Internal_repr)

```

**Comment by diml** [comment]
We need a compilation log in order to be able to provide a diagnostic. I'm closing this issue because there are a lot of issue open and it's hard to follow everything. Please do the following:

- look at the compilation error
- if the output of opam is not enough then do:
  + `opam source core_kernel`
  + `cd core_kernel.<version>`
  + `jbuilder build -p core_kernel`
- then look at the error to see if it's not an obvious issue on your side
- if it's not, open a new issue with the output of `jbuilder build -p core_kernel`m including the version of core_kernel your are trying to install
